### PR TITLE
shim: organize code to different modules

### PR DESCRIPTION
**Pull request by Mossaka**
this commit organized code to different modules:
- constants: to handling constants across entire crate
- engine: the execution engine
- source: manages application sources
- stdio_hook: a trigger hook for stdio
- trigger: manages spin triggers
- utils: utility functions

it also removes ResolvedAppSource as it's not needed

### CODE REVIEW

File: containerd-shim-spin/src/engine.rs [hunk 1]
```diff
@@ -313,349 +221,14 @@ impl SpinEngine {
...
-                .map(|k| k.as_ref())
-                .collect::<Vec<_>>(),
-            ResolvedAppSource::OciRegistry { locked_app } => locked_app
-                .variables
-                .keys()
-                .map(|k| k.as_ref())
-                .collect::<Vec<_>>(),
-        }
-    }
-}
```
Comment (Mossaka) [inline containerd-shim-spin/src/engine.rs:547]: removed entirely cuz it's not needed

**Review by kate-goldenring — APPROVED** [review]
LGTM! Thanks for this reorganization

**Comment by Mossaka** [comment]
Containerd logs for the failured spin wasm workloads:

```
time="2024-09-12T17:36:18.377934221Z" level=error msg="run_wasi ERROR >>>  failed: Unable to listen on 0.0.0.0:80

Caused by:
    Address in use (os error 98)"

```

**Comment by kate-goldenring** [comment]
@Mossaka I wonder if tests that used to run sequentially are now running in parallel or if this is a race case. In the integration tests, maybe we should randomize the host port for port forwarding rather than keeping it always at 8082

**Comment by Mossaka** [comment]
>  In the integration tests, maybe we should randomize the host port for port forwarding rather than keeping it always at 8082

That might be a good idea in general, but it won't help in this case, cuz I can reproduce it in my local running `ctr run` with one instance.

**Comment by Mossaka** [comment]
https://github.com/spinkube/containerd-shim-spin/pull/185#issuecomment-2347496280

is fixed by https://github.com/spinkube/containerd-shim-spin/pull/185/commits/79d54ba559824df85950ef309c8d25472218a432

**Pull request merged**